### PR TITLE
xl support for suite-sparse

### PR DESCRIPTION
**Pull request by eklee15**

### CODE REVIEW

File: var/spack/repos/builtin/packages/suite-sparse/package.py [hunk 1]
```diff
@@ -76,9 +76,12 @@ def install(self, spec, prefix):
             'NVCC          = echo',
             'NVCCFLAGS     =',
         ])
-        if '+fpic' in spec:
+        if '+pic' in spec:
             make_args.extend(['CFLAGS=-fPIC', 'FFLAGS=-fPIC'])
```
Comment (davydden) [inline var/spack/repos/builtin/packages/suite-sparse/package.py]: while fixing this, please call compiler's `pic_flag` property.

Comment (davydden) [inline var/spack/repos/builtin/packages/suite-sparse/package.py]: see https://github.com/LLNL/spack/blob/develop/lib/spack/spack/compilers/clang.py#L117-L119, for example

Comment (davydden) [inline var/spack/repos/builtin/packages/suite-sparse/package.py]: `self.compiler.pic_flag`

Comment (eklee15) [inline var/spack/repos/builtin/packages/suite-sparse/package.py]: What is the right syntax to do it?? 
make_args.extend(['CFLAGS=???', 'FFLAGS=???'])

Comment (davydden) [inline var/spack/repos/builtin/packages/suite-sparse/package.py]:
```
make_args.extend([
    'CFLAGS={0}'.format(self.compiler.pic_flag),
    'FFLAGS={0}'.format(self.compiler.pic_flag)
])
```
should do.

**Comment by eklee15** [comment]
@tgamblin Would you please merge this PR?

**Pull request merged**